app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,48 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def cross_link_amazon_node(new_result)
+      _, az, uid = URI(new_result[:identity_infra]).path.split('/')
+      az_id = AvailabilityZone.find_by(:name => az).try(:id)
+      Vm.where(:uid_ems              => uid,
+               :availability_zone_id => az_id,
+               :type                 => ManageIQ::Providers::Amazon::CloudManager::Vm.name)
+    end
+
+    def cross_link_google_node(new_result)
+      # TODO
+    end
+
+    def cross_link_node(new_result)
+      # Establish a relationship between this node and the vm it is on (if it is in the system)
+      # supported relationships: oVirt, Openstack and VMware.
+      uuid_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                    ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                    ManageIQ::Providers::Vmware::InfraManager::Vm.name]
+
+      # Searching for the underlying instance for Openstack or oVirt.
+      vms = Vm.where(:uid_ems => new_result[:identity_system].downcase, :type => uuid_types)
+      if vms.empty? && new_result[:identity_infra]
```

Comment:
identity_infra is assigned with "try" on line 172, meaning it might be empty. what will happen to this case section if it's empty  - what are the possible values that will come out of line 153?